lib/identity_cache/query_api.rb

```diff
@@ -246,96 +272,102 @@ def find_batch(ids)
         ids.map{ |id| records_by_id[id] }
       end
 
-      def prefetch_associations(associations, records)
-        associations = hashify_includes_structure(associations)
+      def fetch_embedded_associations(records)
+        associations = embedded_associations
+        return true if associations.empty?
```

Comment:
Oh right.  I wasn't quite sure how to handle the case where the parent record has been deleted and can't be refetched.  In many cases the associations have also been deleted, but it is probably safer to fallback to the database for this edge case.  We probably don't need to worry about efficiency for this edge case, so I never ended up using the result.

I'll change these to not explicitly return anything.